### PR TITLE
HPCC-25955 implemented static pvs

### PR DESCRIPTION
**Pull request by rrao4**
Implemented static pvs in order to create storage beyond the eks cluster.
Signed off by raor0000@risk.regn.net

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

I tested the changes using AWS EKS.

### CODE REVIEW

**Comment by xwang2713** [comment]
@g-pan please remove examples/efs/README.md

File: helm/examples/efs/README.md [hunk 1]
```diff
@@ -51,45 +52,114 @@ efs-csi-controller-594c7f67c7-mncgd   3/3     Running   0          32h
...
-Under the helm directory:
+To cleanup, run
 ```console
-  helm install awsstorage examples/efs/hpcc-efs
-  # To start HPCC cluster:
-  helm install myhpcc hpcc/ --set global.image.version=latest -f examples/efs/values-retained-efs.yaml
+  helm uninstall myhpcc
+```
+## 2. Static storage within Kubernetes with values-retained-efs.yaml
+In this method, storage lives on the Kubernetes cluster level. It uses the helm chart "hpcc-efs-dynamic-pv" to manually create PVCs. The PVs are dynamically generated. The PVCs will persist after the HPCC cluster is deleted, meaning that the storage can be reused across different HPCC clusters.<br/>
```
Comment (g-pan) [inline helm/examples/efs/README.md:79]: The PVs are dynamically,  should it be PVCs ??

**Review by g-pan — COMMENTED** [review]
Only one comment/question otherwise approved.

File: helm/examples/efs/create-ap.sh
Comment (xwang2713) [inline helm/examples/efs/create-ap.sh:1]: Add comment that For RISK user tag "owner" and "owner_email" are may required

**Review by xwang2713 — COMMENTED** [review]
Generally looks good. The tags need "owner" and "owner_email" for RISK users.

File: helm/examples/efs/README.md (hunk 1 above)
Comment (rrao4) [reply helm/examples/efs/README.md:79]: No, the PVCs aren't dynamically created, in the hpcc-efs-dynamic-pv/templates/create-pvc.yaml, they are actually manually created.

File: helm/examples/efs/create-ap.sh
Comment (rrao4) [reply helm/examples/efs/create-ap.sh:1]: I added this just now to the README under the explanation of the 3rd storage method.

**Review by g-pan — COMMENTED** [review]
couple of comments in line.

File: helm/examples/efs/README.md [hunk 2]
```diff
@@ -125,3 +196,9 @@ To display the mount targets:
 ```console
   aws efs describe-mount-targets --region <REGION> --file-system-id <EFS ID>
 ```
+## Console
+### Create EFS Server
+Navigate to the EFS service and click "Create file system". Select the same VPC as the one your EKS cluster is running in.
+### Configuring the Mount Targets
+Click on the file system you just created and navigate to "Network". Your mount targets should be displayed.<br/>
+Click on "Manage", and you should see that the security groups for the mount targets are the default security group. Replace these with the EKS cluster security group. To find the EKS cluster security group, navigate to EKS, click on your cluster, click on "Networking", and it is the security group displayed under "Cluster security group". After replacing the default security group for all mount targets, save your changes.
```
[the remark this answers is not in the document]
Comment (rrao4) [reply helm/examples/efs/README.md]: Thanks for the review. I've updated the parts you mentioned.

**Review by g-pan — APPROVED** [review]
approved.

**Review by xwang2713 — APPROVED** [review]
Ready to merge

**Comment by ghalliday** [comment]
Please squash ready for merging

**Comment by rrao4** [comment]
Hi Gavin,

I rebased the commits into one, which is "HPCC-25955 implemented static pvs". Let me know if its ok to merge, or if I should create a new pull request.

Thanks,
Ryan Rao

**Comment by ghalliday** [comment]
I think you need to execute something like:

git fetch upstream
git rebase --onto upstream/candidate-9.2.x HEAD~1

Assuming upstream refers to hpcc-systems/HPCC-Platform

To remove all the other commits from your PR.  I suspect your local candidate-9.2.x branch is out of sync with the hpcc-systems branch.

**Comment by rrao4** [comment]
Closed, new PR #17687 has been created with the correct base and same changes.

**Pull request closed without merging**